### PR TITLE
changed for loops with int vs long comparison

### PR DESCRIPTION
**Pull request by macfarla**
Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

Where int was being compared with long. Mostly int being used to iterate over values which could be long.
Not sure about these 2 since they are dealing with uint
- FlexiblePrivacyGroupContract
- FlexiblePrivacyGroupController

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/FlexiblePrivacyController.java [hunk 1]
```diff
@@ -205,7 +205,7 @@ private List<String> decodeParticipantList(final Bytes rlpEncodedList) {
     final ArrayList<String> decodedElements = new ArrayList<>();
     // first 32 bytes is dynamic list offset
     final UInt256 lengthOfList = UInt256.fromBytes(rlpEncodedList.slice(32, 32)); // length of list
-    for (int i = 0; i < lengthOfList.toLong(); ++i) {
+    for (int i = 0; i < lengthOfList.toInt(); ++i) {
```
[the remark this answers is not in the document]
Comment (macfarla) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/FlexiblePrivacyController.java]: I tried without the UInt256.fromBytes but it didn't work. I did get rid of the duplication though.

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/FlexiblePrivacyGroupContract.java [hunk 2]
```diff
@@ -229,12 +228,13 @@ private CallParameter buildCallParams(final Bytes methodCall) {
...
-  private List<String> decodeList(final Bytes rlpEncodedList) {
+  public static List<String> decodeList(final Bytes rlpEncodedList) {
     final ArrayList<String> decodedElements = new ArrayList<>();
     // first 32 bytes is dynamic list offset
     if (rlpEncodedList.size() < 64) return decodedElements;
-    final long lengthOfList =
-        UInt256.fromBytes(rlpEncodedList.slice(32, 32)).toLong(); // length of list
+    // Bytes uses a byte[] for the content which can only have up to Integer.MAX_VALUE-5 elements
+    final int lengthOfList =
+        UInt256.fromBytes(rlpEncodedList.slice(32, 32)).toInt(); // length of list
```
Comment (pinges) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/FlexiblePrivacyGroupContract.java:237]: nit: please remove the UInt256.fromBytes, as you can do the toInt() on the slice (Bytes).

**Review by pinges — APPROVED** [review]
please check my comment!

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/FlexiblePrivacyGroupContract.java (hunk 2 above)
Comment (macfarla) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/FlexiblePrivacyGroupContract.java:237]: as discussed, this needs to stay :)

Comment (macfarla) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/FlexiblePrivacyGroupContract.java:237]: Value of size 32 has more than 4 bytes
Without this, running tests I get this error: 
java.lang.IllegalArgumentException: Value of size 32 has more than 4 bytes
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:193)
	at org.apache.tuweni.bytes.Bytes.toInt(Bytes.java:685)
	at org.apache.tuweni.bytes.Bytes.toInt(Bytes.java:673)
	at org.hyperledger.besu.ethereum.privacy.FlexiblePrivacyGroupContract.decodeList(FlexiblePrivacyGroupContract.java:236)

**Pull request merged**